octavia-cli/octavia_cli/apply/resources.py

```diff
@@ -599,15 +652,21 @@ def _check_for_legacy_connection_configuration_keys(self, configuration_to_check
             )
 
     def _get_remote_comparable_configuration(self) -> dict:
-        keys_to_filter_out = [
-            "name",
-            "source_id",
-            "destination_id",
-            "connection_id",
-            "operation_ids",
-            "source_catalog_id",
-        ]  # We do not allow local editing of these keys
-        return {k: v for k, v in self.remote_resource.to_dict().items() if k not in keys_to_filter_out}
+
+        comparable = {
+            k: v for k, v in self.remote_resource.to_dict().items() if k not in self.remote_root_level_keys_to_filter_out_for_comparison
+        }
+        if "operations" in comparable:
+            for operation in comparable["operations"]:
+                for k in self.remote_operation_level_keys_to_filter_out:
+                    operation.pop(k)
+                if "dbt" in operation["operator_configuration"] and operation["operator_configuration"]["dbt"] is None:
+                    operation["operator_configuration"].pop("dbt")
+                if "normalization" in operation["operator_configuration"] and operation["operator_configuration"]["normalization"] is None:
+                    operation["operator_configuration"].pop("normalization")
```

Comment:
I'm not understanding why we need all of this special logic for operations in this method. Could you elaborate on why this is necessary, and maybe leave a comment here explaining what is going on?

Reply:
The payload returned by the API with the operator might contain null values for `dbt` field (when the operator configuration is normalization) or the `normalization` field (when the operator configuration is DBT and normalization is disabled). I remove these keys from the remote config to compare them properly to the local config.

Reply:
@alafanechere I just wanted to double check: now that we have made the API change to exclude null values, is this logic still necessary?